### ISSUE TITLE
Grafana table column ordering changes on refresh

### ISSUE THREAD

**Issue by Diptarko** [grafana/grafana#71260]
### 
# What went wrong?

Ability to keep table column  ordering upon refresh
What happened:
We have a dashboard with a table panel .The columns come in a particular order, . The next time Grafana executed the query (the dashboard has 10s refresh) the columns reordered, and the column I was trying to look at now was now on the opposite side of the table.

What you expected to happen:
Table columns to stay in the same place even with a page refresh.

How to reproduce it (as minimally and precisely as possible):
Load a dashboard with a table panel, use the refresh button in the top right and watch the columns reorder.

Anything else we need to know?:

Environment:

Grafana version: Grafana v9.4.3

Data source type & version: InfluxDB v2.7.1
User OS & Browser: Windows 10/Ubuntu , Chrome
**What did you expect to happen**:
The columns should stay in the correct order.
  - 

### 
# How do we reproduce it?

Create a Grafana table panel and populate it with data and create columns
Refresh the dashboard to every 10 S
Check if the table column order is getting changed.

### 
# What Grafana version are you using?

V9.4.3

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

None

### Is this a Regression?

None

**Comment by zuchka** [comment]
do you think you could try and replicate this on our public sandbox, https://play.grafana.org? :pray:

**Comment by braeluca** [comment]
I also have the same issue. I could reproduce it on the public sandbox. All you need is a table with the following flux query:

```
import "generate"

generate.from(
    count: 6,
    fn: (n) => (n + 1) * (n + 2),
    start: 2021-01-01T00:00:00Z,
    stop: 2021-01-02T00:00:00Z,
) |> map(fn: (r) => ({ _value:r._value, _time:r._time, _field: "_field", _tag: "_tag"}))
```

If you then spam the refresh dashboard button, it happends quite often, that the columns are swapped randomly. I also attached screenshots where this could be seen:

![grafana_table_bug_1](https://github.com/grafana/grafana/assets/43317113/06c4baa5-f1a3-49f1-8e6a-c0c661005e06)

![grafana_table_bug_2](https://github.com/grafana/grafana/assets/43317113/d2c17103-d1d7-4b54-9cd0-7b457706edd1)

![grafana_table_bug_3](https://github.com/grafana/grafana/assets/43317113/6d941741-0480-4972-8f10-efcb7d334db1)

 Could you verify this behavior?

**Comment by NPetersenDK** [comment]
I have the same issue with flux and ordering changes on refresh. Have you guys found a solution, or are you still waiting on Grafana to fix it / say why it happens?

**Comment by konfilios** [comment]
This is happening on `Grafana v10.4.2`. When the results of an influxdb (influxql, not flux) query are formatted as `Table` then each refresh moves columns around.

That might sound like just an annoying cosmetic problem, but it's worse: If you try to do any transformations (e.g. `Grouping to Matrix`) then they work for a few seconds and then break.

Not sure what I need to do to fix the order of the columns.

**Comment by yurividal** [comment]
Has a workarround for this ever been found?

**Comment by NPetersenDK** [comment]
@yurividal not that i know of, we still see the issue atleast.

**Comment by braeluca** [comment]
@yurividal @NPetersenDK For me worked using the Organize fields by name transformation. I had this before but somehow it only works now since I reapplied the transformation. But for now I not have seen that the column ordering has changed.

**Comment by yurividal** [comment]
> [@yurividal](https://github.com/yurividal) [@NPetersenDK](https://github.com/NPetersenDK) For me worked using the Organize fields by name transformation. I had this before but somehow it only works now since I reapplied the transformation. But for now I not have seen that the column ordering has changed.

Yes, this worked for me too! The transformation was already there , but I had to remove it and add it again, now it works!

**Comment by jwiedmann-vdura** [comment]
@NPetersenDK I'm running Grafana v12.1.0 and found that I need to put two transitions. The first is an "Organize fields by name", set to Auto and ASC. That fixed the columns from jumping around, but in the wrong order. Then I placed another "Organize fields by name" and set the column order as I wanted. It's greasy, but it works.

**Comment by ShadowJonathan** [comment]
https://nostalebots.xyz/
